node/src/exchange.rs

```diff
@@ -31,21 +34,101 @@ pub fn new_behaviour(network: &str) -> Behaviour {
 #[derive(Clone, Copy, Debug, Default)]
 pub struct HeaderCodec;
 
+impl HeaderCodec {
+    async fn read_raw_message<T, R>(
```

Comment:
single we return a decoded message, the `raw` is a misleading. I suggest this:

```rust
fn read_raw_message<R, T>(reader: &mut R, buf: &mut Vec<u8>, max_len: usize) -> io::Result<Option<T>>
```

Reply:
renamed the fn and generic types to match